include/cliargsparser.h

```diff
@@ -72,6 +72,12 @@ class CliArgsParser {
 
 	nonstd::optional<std::string> config_file() const;
 
+	nonstd::optional<std::string> queue_file() const;
+
+	nonstd::optional<std::string> search_file() const;
+
+	nonstd::optional<std::string> cmdline_file() const;
```

Comment:
How about naming these `search_history_file()` and `cmdline_history_file()` for clarity?